model/include/model/plugin_handler.h

```diff
@@ -189,9 +189,31 @@ class PluginHandler {
   bool InstallPlugin(const std::string& path, std::string& filelist,
                      const std::string metadata_path, bool only_metadata);
 
+  /**
+   * Internal helper function to extract a tarball into platform-specific user directories.
+   *
+   *   @param path: Path to tarball.
+   *   @param filelist: On return contains a list of files installed.
+   *   @param metadata_path: If non-empty, location where to extract plugin metadata.xml file.
+   *   @param only_metadata: If true don't install any files, just extract
+   *                         the metadata.xml file.
+   *   @return true if tarball contains metadata.xml file.
```

Comment:
Previously, this function returned true if metadata_path == "" or tarball contains metadata.xml file. I.e. the function would return true for a tarball that does not contain the `metadata.xml` file, as long as metadata_path == "". I feel this is not very intuitive. It is easier to understand if the function returns true if the tarball can be extracted and it contains the metadata.xml file.

Reply:
Yes, the current semantics are historic, at start we did not bundle the metadata in the plugin and thus needed to accept such plugins in a transition window.

Generally, you don't really cope with malformed XML. From an overall perspective, this should probably return true if the metadata file exists and can be parsed.